compiler/rustc_ast_lowering/src/item.rs

```diff
@@ -928,7 +920,7 @@ impl<'hir> LoweringContext<'_, 'hir> {
         let has_value = true;
         let (defaultness, _) = self.lower_defaultness(i.kind.defaultness(), has_value);
         hir::ImplItemRef {
-            id: hir::ImplItemId { def_id: self.lower_node_id(i.id).expect_owner() },
+            id: hir::ImplItemId { def_id: self.allocate_hir_id_counter(i.id) },
```

Comment:
I'm not sure why this difference between `lower_trait_item_ref` and `lower_impl_item_ref` exists.
For both of them `allocate_hir_id_counter` should have already been called by `visit_assoc_item`?

Reply:
`lower_impl_item_ref` is called as part of `lower_item` for the impl block. At that time, `allocate_hir_id_counter` has not been called yet for the impl_item. However, this `allocate_hir_id_counter` is required for the call to `lower_visibility`. This difference is removed by https://github.com/rust-lang/rust/pull/87234 as it removes the call to ``lower_visibility`.